src/test/test-vector.c

```diff
@@ -360,4 +361,51 @@ vector_initializes_elements_to_zero(void)
 }
 
 
+DEFINE_TEST(test_vector_del_element);
+
+static void
+test_vector_del_element(void)
```

Comment:
please also add a test to `sol_ptr_vector_del_element()`, I think your wrapper is not correct.